toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt

```diff
@@ -468,6 +490,20 @@ public class TraceState(
         _currentTraceName.value = name
     }
 
+    internal fun selectNextCompletedTrace() {
+        val selectedTraceRunIndex = completedTraces.indexOf(_selectedTraceRun.value)
+        if (selectedTraceRunIndex + 1 < _completedTraces.size) {
+            _selectedCompletedTraceIndex = selectedTraceRunIndex + 1
+        }
+    }
```

Comment:
This could be simplified with only keeping track of the index: 
```suggestion
    internal fun selectNextCompletedTrace() {
        if (currentTraceResultIndex.value + 1 < _completedTraces.size) {
            currentTraceResultIndex.value += 1
        }
    }
```